.github/workflows/psalm-taint-analysis.yml

```
@@ -0,0 +1,77 @@
+name: psalm taint analysis
+
+on:
+  workflow_call:
+    inputs:
+      report-retention-days:
+        description: Duration in days to preserve reports.
+        required: false
+        default: "5"
+        type: string
+      working-directory:
+        description: "Workflow working directory"
+        default: '.'
+        required: false
+        type: string
+      php-version:
+        description: 'Setup PHP version.'
+        default: '8.1'
+        required: true
+        type: string
+      extensions:
+        description: 'Setup PHP extensions.'
+        required: false
+        type: string
+    secrets:
+      gpr-key:
+        description: Personal access token to fetch private repos
+        required: false
+
+jobs:
+  psalm-taint-analysis:
+    name: psalm taint analysis
+    runs-on: ubuntu-latest
+    env:
+      cache-key: v1
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup cache environment
+        id: extcache
+        uses: shivammathur/cache-extensions@v1
+        with:
+          php-version: ${{ inputs.php-version }}
+          extensions: ${{ inputs.extensions }}
+          key: ${{ env.cache-key }}
+
+      - name: Cache extensions
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.extcache.outputs.dir }}
+          key: ${{ steps.extcache.outputs.key }}
+          restore-keys: ${{ steps.extcache.outputs.key }}
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ inputs.php-version }}
+          extensions: ${{ inputs.extensions }}
+
+      - name: Composer install
+        uses: equisoft-actions/composer@442ba88f61817ea57e7fda46cfbc8664fa14ceeb
+        with:
+          gpr-key: ${{ secrets.gpr-key }}
+          working-directory: ${{ inputs.working-directory }}
+
+      - name: psalm taint-analysis
+        working-directory: ${{ inputs.working-directory }}
+        shell: bash
+        run: ./vendor/bin/psalm --taint-analysis --report=psalm-taint-analysis.sarif
+
+      - name: Store results.sarif
+        uses: actions/upload-artifact@v2
+        with:
```

Est-ce que psalm exit avec autre chose que 0 en cas d'échec? Ça causerait le sarif de ne pas être uploadé.

```suggestion
        uses: actions/upload-artifact@v2
        if: "!cancelled()"
        with:
```